privacyidea.py

```diff
@@ -2,131 +2,241 @@
-            # TODO: Send the password and the transaction_id to privacyIDEA.
-            # Pass the user into the session.
-            if user_password == "otp":
-                return True
+            if user_name:
+                self.addToSession("currentUser", user_name)
+
+                response = self.pi.validateCheck(user_name, user_password)
+                if response:
+                    # First check if what was entered is sufficient to log in
+                    if response.getValue():
+                        logged_in = authenticationService.authenticate(user_name)
+                        if logged_in:
+                            self.addToSession("auth_success", True)
+                        return logged_in
+
+                    # If not, check if transaction was triggered
+                    if response.getTransactionID():
+                        identity.setWorkingParameter("transaction_message", response.getMessage())
+                        self.addToSession("transaction_id", response.getTransactionID())
+
+                        # Check if push is available
+                        tttList = response.getTriggeredTokenTypes()
+                        if tttList.contains("push"):
+                            identity.setWorkingParameter("push_available", "1")
+                            tttList.remove("push")                        
+                        
+                        # Check if an input field is needed for any other token type
+                        if tttList.size() > 0:
+                            identity.setWorkingParameter("otp_available", "1")
+
+                        return True
+                else:
+                    print("privacyIDEA. Empty response from server")
+                    fm.add(FacesMessage.SEVERITY_ERROR, "No response from the privacyIDEA Server. Please check the connection!")
+
+                return False
+            else:
+                #print("privacyIDEA. Username is empty")
+                fm.add(FacesMessage.SEVERITY_ERROR, "Please enter a username!")
 
-    def prepareForStep(self, configurationAttributes, requestParameters, step):
-        if step == 1:
-            print "privacyIDEA. Prepare for Step 1"
-            return True
-        elif step == 2:
-            # This is a 2nd challenge response step
-            print("privacyIDEA. Prepare for another Step {0!s}.".format(step))
-            return True
-        else:
             return False
+        else:
+            # Get the user from step 1
+            currentUser = self.getFromSession("currentUser")
+
+            if currentUser:
+                authenticationService.authenticate(currentUser)
+            else:
+                print("privacyIDEA. No user found in session for second step")
+                fm.add(FacesMessage.SEVERITY_ERROR, "An error occurred. Please try to restart the authentication!")
```

Comment:
Should we say something like, that "session data got lost"?

Reply:
Ok